drivers/adc/Kconfig.adc_emul

```diff
@@ -0,0 +1,27 @@
+# Copyright 2021 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+config ADC_EMUL
+	bool "ADC emulator"
```

Comment:
Should this have a `depends on SOC_POSIX` or `depends on ARCH_POSIX`?

Reply:
I think it is not necessary, because ADC emulator doesn't use POSIX specific functions.